### PR TITLE
Fixed WebUI crash when a status opened in the media modal is deleted

### PR DESCRIPTION
**Pull request by kaias1jp**
Fixed a problem that a compatibility error occurs when the status is deleted while the status image etc. is selected and displayed.

### CODE REVIEW

**Comment by ClearlyClaire** [comment]
Thank you for the contribution! I took the liberty of changing the PR title to reflect that it's not a compatibility error but just a crash.

Your change seem to fix the crash itself, but it would result in the pop-out player still being displayed, although without the status bar. A better approach would probably be to close the pop-out player as soon as the status is deleted.

This could be handled by changing `app/javascript/mastodon/reducers/picture_in_picture.js` to process status deletion the same way as `PICTURE_IN_PICTURE_REMOVE` when the deleted status id is the currently-displayed one. Statuses are deleted through a `TIMELINE_DELETE` action (see `app/javascript/mastodon/reducers/statuses.js` for example).

**Comment by kaias1jp** [comment]
Thank you. You're right. I will consider how to close the pop-out player, but should this PR be closed once? Since this was my first PR, I don't know what to do in such a case. If you do not mind, please teach me.

**Comment by ClearlyClaire** [comment]
You can push additional commits in the PR. I'd start by reverting the latest commit (`git revert HEAD`) and continue working as before.

This will result in a slightly noisy commit history, but that isn't really an issue. If you still want to clean that commit history later on, you can use `git rebase -i`, but that is completely optional.

**Comment by kaias1jp** [comment]
I reconfirmed where the crash occurred.
What I found was that the WebUI crashed when the status was deleted when the image was displayed modally.
So I want to change the title to "Fixed WebUI crashing when status is deleted when displaying images modally".
As an additional fix, I would like to close the image only if the modally displayed image has a deleted status.
The pop-out player does not crash, but I think it is necessary to close the player by deleting the status. I'm wondering whether to include this in this PR or put it out as another PR.

**Comment by ClearlyClaire** [comment]
Ah yes, there is indeed a similar crash in the media modal. I'd say it's a similar issue but not exactly the same. I don't have a preference with regards to fixing both issues in the same PR or different ones, do what's easiest for you.

File: app/javascript/mastodon/reducers/modal.js [hunk 1]
```diff
@@ -11,6 +12,12 @@ export default function modal(state = initialState, action) {
     return { modalType: action.modalType, modalProps: action.modalProps };
   case MODAL_CLOSE:
     return (action.modalType === undefined || action.modalType === state.modalType) ? initialState : state;
+  case TIMELINE_DELETE:
+    if (state.modalProps.statusId === action.id) {
+      return (action.modalType === undefined || action.modalType === state.modalType) ? initialState : state;
+    } else {
+      return state;
+    }
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/reducers/modal.js:20]: This action never has a `modalType` attribute, so you can simply do `return (state.modalProps.statusId === action.id) ? initialState : state;` here!

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

File: app/javascript/mastodon/reducers/modal.js (hunk 1 above)
Comment (kaias1jp) [reply app/javascript/mastodon/reducers/modal.js]: Thank you. I fixed the code.

**Review by ClearlyClaire — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
Looks good to me, thanks!

**Review by Gargron — APPROVED** [review]

**Pull request merged**